src/repl/interfaces.ts

```diff
@@ -45,3 +45,8 @@ export interface IReplExtensionFactory {
     // Called during REPL initialization.
     create(world: World, dataPath: string): IReplExtension;
 }
+
+// tslint:disable-next-line:interface-name
+export interface IRepl {
```

Comment:
The whole PR looks good from a design perspective. One small detail is that we probably want IRepl to be backed by a class, rather than being a plain old javascript object (POJO). It might make sense to do the following

1. Change the `server` property to a function that returns a replServer.REPLServer.
2. Make a Repl class that implements IRepl.

The reason I am recommending this is that we will be adding other methods to IRepl in the very near future. Since some portions of the interface will be methods, we should probably make server be a method as well.